### PR TITLE
twitter-text: Add several missing APIs and tests for them

### PR DESCRIPTION
**Pull request by rhysd**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/twitter/twitter-text/tree/master/js/src
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

While I was reading the implementation of twitter-text package, I noticed several APIs are missing in its type definitions. So I added the types reading its implementation because there is no good documentation of twitter-text package.

This PR adds the missing several APIs and tests for them.

### CODE REVIEW

**Comment by rhysd** [comment]
How can I get review from maintainer? More than 2 weeks passed so far.

**Review by jakebailey — APPROVED** [review]

**Comment by rhysd** [comment]
Ready to merge

**Pull request merged**
